java/arcs/core/host/Utils.kt

```diff
@@ -17,3 +22,19 @@ fun KClass<*>.className(): String {
         .substringBefore('<')
         .replace('$', '.')
 }
+
+fun Type.toSchema(): Schema {
+    when (this) {
+        is SingletonType<*> -> if (this.containedType is EntityType) {
+            return (this.containedType as EntityType).entitySchema
+        }
+        is CollectionType<*> -> if (this.collectionType is EntityType) {
+            return (this.collectionType as EntityType).entitySchema
+        }
+        is EntityType -> return this.entitySchema
+        else -> Unit
+    }
+    throw Exception("Can't get entitySchema of unknown type $this")
```

Comment:
Please throw a more descriptive Exception class.. I think something like "UnsupportedOperationException" or "IllegalStateException" might be better here.